test/SqlExtension.Tests/IntegrationTestBase.cs

```diff
@@ -0,0 +1,252 @@
+        protected string DatabaseName { get; private set; }
+
+        /// <summary>
+        /// Output redirect for XUnit tests.
+        /// Please use TestOutput.WriteLine() instead of Console or Debug.
+        /// </summary>
+        protected ITestOutputHelper TestOutput { get; private set; }
+
+        protected int Port { get; private set; }
+
+        public IntegrationTestBase(ITestOutputHelper output)
+        {
+            TestOutput = output;
+
+            SetupDatabase();
+            
+            StartFunctionHost();
+        }
+
+        /// <remarks>
+        /// Integration tests depend on a localhost server to be running.
+        /// Either have one running locally with integrated auth, or start a mssql instance in a Docker container 
+        /// and set "SA_PASSWORD" as environment variable before running "dotnet tets".
+        /// </remarks>
+        private void SetupDatabase()
+        {
+            // First connect to master to create the database
+            SqlConnectionStringBuilder connectionStringBuilder = new SqlConnectionStringBuilder()
+            {
+                DataSource = "localhost",
+                InitialCatalog = "master",
+                Pooling = false
+            };
+
+            string userId = "SA";
+            string password = Environment.GetEnvironmentVariable("SA_PASSWORD");
+            if (string.IsNullOrEmpty(password))
+            {
+                connectionStringBuilder.IntegratedSecurity = true;
+            }
+            else
+            {
+                connectionStringBuilder.UserID = userId;
+                connectionStringBuilder.Password = password;
+            }
+
+            // Create database
+            DatabaseName = TestUtils.GetUniqueDBName("SqlBindingsTest");
+            using (SqlConnection masterConnection = new SqlConnection(connectionStringBuilder.ToString()))
+            {
+                masterConnection.Open();
+                TestUtils.ExecuteNonQuery(masterConnection, $"CREATE DATABASE [{DatabaseName}]");
+            }
+
+            // Setup connection
+            connectionStringBuilder.InitialCatalog = DatabaseName;
+            Connection = new SqlConnection(connectionStringBuilder.ToString());
+            Connection.Open();
+
+            // Create the database definition
+            // Ideally all the sql files would be in a sqlproj and can just be deployed
+            string databaseScriptsPath = Path.Combine(GetPathToSamplesBin(), "Database");
+            foreach (string file in Directory.EnumerateFiles(databaseScriptsPath, "*.sql"))
+            {
+                ExecuteNonQuery(File.ReadAllText(file));
+            }
+
+            // Set SqlConnectionString env var for the Function to use
+            Environment.SetEnvironmentVariable("SqlConnectionString", connectionStringBuilder.ToString());
+        }
+
+        private void StartFunctionHost()
+        {
+            Port = 7071 + TestUtils.ThreadId;
```

Comment:
Why do you need to increase this? Shouldn't there only be one function host running at a time?

Reply:
Removed. Left over code from when I attempted to multithread it